src/main/java/com/erigitic/commands/JobCommand.java

```diff
@@ -25,67 +25,264 @@
+                    .executor(new Set(totalEconomy))
+                    .arguments(
+                            GenericArguments.string(Text.of("jobName")),
+                            GenericArguments.requiringPermission(
+                                    // Use "User" instead of "Player" so we achieve offline support
+                                    GenericArguments.userOrSource(Text.of("user")),
+                                    TEPermissions.JOB_SET_OTHERS
+                            )
+                            //TODO: Maybe a WEAK/STRONG param for setting an others job AND ignoring requirements
+                    )
+                    .build();
+        }
+
+        private TotalEconomy totalEconomy;
+
+        public Set(TotalEconomy totalEconomy) {
+            this.totalEconomy = totalEconomy;
+        }
+
+        @Override
+        public CommandResult execute(CommandSource src, CommandContext args) throws CommandException {
+            //TODO: Do checks here, in case we or other plugins want to bypass them in the future
+
+            //JobName should always be present: Enforced by Sponge
+            String jobName = args.getOne("jobName").get().toString().toLowerCase();
+            //Player also enforced by Sponge
+            User user = args.<User>getOne("user").get();
+
+            Optional<TEJob> optJob = totalEconomy.getTEJobManager().getJob(jobName, false);
+            if (!optJob.isPresent()) throw new CommandException(Text.of("Job " + jobName + " does not exist!"));
+
+            TEJob job = optJob.get();
+             if (job.getRequirement().isPresent()) {
+                 JobBasedRequirement req = job.getRequirement().get();
 
-            // Do checks here, in case we or other plugins want to bypass them in the future
-            if (args.getOne("jobName").isPresent()) {
-                String jobName = args.getOne("jobName").get().toString().toLowerCase();
+                 if (req.permissionNeeded() != null && !user.hasPermission(req.permissionNeeded()))
+                     throw new CommandException(Text.of("Not permitted to join job \"", TextColors.YELLOW, jobName, TextColors.RED, "\""));
 
-                Optional<TEJob> optJob = teJobManager.getJob(jobName, false);
-                if (!optJob.isPresent()) throw new CommandException(Text.of("Job " + jobName + " does not exist!"));
+                 if (req.jobNeeded() != null && req.jobLevelNeeded() > totalEconomy.getTEJobManager().getJobLevel(req.jobNeeded().toLowerCase(), user)) {
+                     throw new CommandException(Text.of("Insufficient level! Level ",
+                             TextColors.YELLOW, req.jobLevelNeeded(), TextColors.RED," as a ",
+                             TextColors.YELLOW, req.jobNeeded(), TextColors.RED, " required!"));
+                 }
+             }
 
-                TEJob job = optJob.get();
+            src.sendMessage(Text.of(TextColors.YELLOW, "Changing job..."));
```

Comment:
There's no feedback to the ``CommandSource`` when running ``job set <job> <player>`` without this.